scripts/neighbor_advertiser

```diff
@@ -281,32 +294,58 @@ def construct_neighbor_advertiser_slice():
 
     all_vlan_interfaces = get_vlan_interfaces()
 
+    vlan_intf_table = config_db.get_table('VLAN_INTERFACE')
+
+    vxlanPort = appl_db.get_table('SWITCH_TABLE:switch')['vxlan_port']
+
     for vlan_interface in all_vlan_interfaces:
-        vlan_id, vxlan_id, ipv4_addr, ipv6_addr, mac_addr = get_vlan_addresses(vlan_interface)
+        vlan_id, vxlan_id, ipv4_addr, ipv4_prefix, ipv6_addr, ipv6_prefix, mac_addr = get_vlan_addresses(vlan_interface)
 
         if not mac_addr:
             log.log_warning('Cannot find mac addr of vlan interface {}'.format(vlan_interface))
             continue
 
         ipv4_mappings = []
+        ipv6_mappings = []
+        
+        ctr = 0
         for addr in ipv4_addr:
+            if 'proxy_arp' in vlan_intf_table[vlan_interface]:
+                if vlan_intf_table[vlan_interface]['proxy_arp'] == 'enabled':
```

Comment:
is it better to check with an "and" operator? Just a suggestion to avoid extra "else"